### PR TITLE
Rename isASCII to isAscii

### PR DESCRIPTION
**Pull request by mjbshaw**
"ASCII" should be treated as a normal word according to [the style guide](https://ziglang.org/documentation/0.13.0/#Names), and thus `isASCII` should be cased as `isAscii`.

This keeps `isASCII` as a deprecated alias for `isAscii` to ease the transition for users.

### CODE REVIEW

**Comment by nektro** [comment]
LGTM but while we're at is maybe we could apply [Avoid Redundant Names in Fully-Qualified Namespaces](https://ziglang.org/documentation/master/#Avoid-Redundant-Names-in-Fully-Qualified-Namespaces) and make it `std.ascii.is()` ?

**Comment by ifreund** [comment]
Since we are bikeshedding naming, I personally would prefer `std.ascii.valid()` to std.ascii.is()`. I think it reads better and that the meaning is more obvious. Compare the following:

```zig
if (std.ascii.is("abc")) foobar();

if (std.ascii.valid("abc")) foobar();
```

**Comment by mjbshaw** [comment]
I'll change it to whatever Zig maintainers want, but here are my thoughts on the two suggested names:

`std.ascii.is()` takes it too far and loses the clarity the original name had. When I read this, my immediate reaction is "Is what? Whitespace? Printable? Any valid ASCII character? etc." The name `is()` is too nondescript to be useful.

`std.ascii.valid()` is okay but not great. There are plenty of other Zig functions that start with `is` when the function is basically asking a question. At the very least it should be `isValid()`, which would be more consistent with other `isValidX()` functions in std like [`std.zig.isValidId()`](https://ziglang.org/documentation/master/std/#std.zig.isValidId), [`std.net.isValidHostName()`](https://ziglang.org/documentation/master/std/#std.net.isValidHostName), [`std.mem.isValidAlign()`](https://ziglang.org/documentation/master/std/#std.mem.isValidAlign), and others.

That said, I think even `isValid()` isn't particularly great. `isAscii()` is no longer than `isValid()`, but it's *immediately* clear to me what `isAscii()` does just from its name. `isValid()` doesn't do that; I need its parent scope (`ascii.isValid()`) to at least begin to guess what it does.

So my preference would be to keep this function named `isAscii()`. It's immediately clear what it does, which is extremely valuable. It is admittely slightly repetetive when used as `std.ascii.isAscii()`, but it's not any longer or noisier than `std.ascii.isValid()`. I think the spirit of that "avoid redundant names" guidance is to cut down on the noise. `isValid()` doesn't cut down on any noise compared to `isAscii()`.

But like I said, I'll defer to the maintainers, so please let me know if a name change is needed.

**Comment by mjbshaw** [comment]
That would be a large departure from many of `std`'s current functions, including the ones I previously linked to and many `std.math.is*` functions. I disagree with that view; I'm happy with Zig's current naming approach. I don't want to evolve this PR into a massive rename.

My goal with this PR is quite simple: bring `isASCII()` into conformance with Zig's existing style guide.

**Review by ifreund — APPROVED** [review]
Yeah, let's cut the bikeshedding short here and move on. This is a fine incremental improvement to the status quo and I don't think any of the alternatives discussed are objectively better.

**Comment by ifreund** [comment]
Thanks for the PR!

**Pull request merged**
